types/check.go

```diff
@@ -72,6 +71,9 @@ func NewCheck(c *CheckConfig) *Check {
 		OutputMetricHandlers: c.OutputMetricHandlers,
 		EnvVars:              c.EnvVars,
 	}
+	dummyCheck := &Check{}
+	_ = dynamic.Unmarshal(c.ExtendedAttributes, dummyCheck)
+	check.ExtendedAttributes = dummyCheck.ExtendedAttributes
```

Comment:
Kill dummyCheck?

Reply:
No, this fixes the bug.

Reply:
Can we name this something else?

Reply:
Sure, although I'm having trouble coming up with something better. Any suggestions?